.github/workflows/web-demo.yml

```diff
@@ -0,0 +1,66 @@
+env:
+  # We aim to always test with the latest stable Rust toolchain, however we pin to a specific
+  # version like 1.70. Note that we only specify MAJOR.MINOR and not PATCH so that bugfixes still
+  # come automatically. If the version specified here is no longer the latest stable version,
+  # then please feel free to submit a PR that adjusts it along with the potential clippy fixes.
+  RUST_STABLE_VER: "1.76" # In quotes because otherwise (e.g.) 1.70 would be interpreted as 1.7
+
+name: Web Demo Release
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  rustfmt:
+    runs-on: ubuntu-latest
+    name: cargo fmt
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: install stable toolchain
+        uses: dtolnay/rust-toolchain@master
+        with:
+          toolchain: ${{ env.RUST_STABLE_VER }}
+          targets: wasm32-unknown-unknown
+
+      - name: install wasm-bindgen
+        uses: jetli/wasm-bindgen-action@v0.2.0
+        with:
+          version: 'latest'
+
+      - name: build (wasm)
+        run: cargo build -p with_winit --bin with_winit_bin --release --target wasm32-unknown-unknown
+        env:
+          RUSTFLAGS: '--cfg=web_sys_unstable_apis'
+
+      - name: package wasm
+        run: |
+          mkdir public
+          wasm-bindgen --target web --out-dir public target/wasm32-unknown-unknown/release/with_winit_bin.wasm --no-typescript
+          cat << EOF > public/index.html
+            <html>
+              <title>Vello Web Demo</title>
+              <meta content=no-cache http-equiv=Cache-control>
+              <meta content=-1 http-equiv=Expires>
+              <script type=module>import initSync from"/vello/with_winit_bin.js";initSync(`/vello/with_winit_bin_bg.wasm`);</script>
+              <link as=fetch crossorigin href=/vello/with_winit_bin_bg.wasm rel=preload type=application/wasm>
+              <link crossorigin href=/vello/with_winit_bin.js rel=modulepreload>
+              </head>
+              <body>
+                <style>
+                  body {
+                    margin: 0;
+                    padding: 0;
+                  }
+                </style>
+              </body>
+            </html>
+          EOF
+
+      - name: deploy
+        uses: peaceiris/actions-gh-pages@v3
```

Comment:
Why is this not using https://github.com/marketplace/actions/deploy-github-pages-site? (i.e. the official one from GitHub). We at least need a comment here explaining that choice

Reply:
Added a comment TODO to revisit when stable. While `actions/deploy-pages` is stable, they require you to use [`actions/upload-pages-artifact`](https://github.com/actions/upload-pages-artifact) which is not stable.

Reply:
I think I'd still much rather us use the official GitHub one - creating a gh-pages branch is messy, and requires additional tokens to be managed.

Like yes, there's risk that things break, but needing to do secrets securely is something I don't want to deal with at the moment

Reply:
Do you mean `secrets.GITHUB_TOKEN`? This is a token provided to all jobs by GitHub, not something needing to be added or managed by us.

I can still work on the official one, though. I'll have it up soon.

Reply:
Ah, thanks for the pointer. I'd never dug into it before. My prior experience was with talking to external systems, which of course do need tokens managing.

Even still, I do think it's nicer to not make a gh-pages branch, so thanks for changing that